src/components/Popup/index.tsx

```diff
@@ -0,0 +1,23 @@
+// src/components/Popup/index.tsx
+import React from 'react';
+import styles from './index.module.scss';
+
+interface PopupProps {
+  onClose: () => void;
+  content: JSX.Element; // Accept a JSX element as content
+}
+
+const Popup: React.FC<PopupProps> = ({ onClose, content }) => {
+  return (
+    <div className={styles.popupContainer}>
+      <div className={styles.popupContent}>
+        {content}
+        <button className={styles.closeBtn} onClick={onClose}>
```

Comment:
IMO the Popup must handle close itself. and also allows the parent to pass the callback of on close into